tabular/tests/unittests/autogluon/tabular/configs/test_config_helper.py

```diff
@@ -0,0 +1,284 @@
+import numpy as np
+import pytest
+from sklearn.feature_extraction.text import CountVectorizer
+
+from autogluon.features import TextNgramFeatureGenerator
+from autogluon.tabular.configs.config_helper import ConfigBuilder
+
+
+def test_presets():
+    expected_config = dict(presets=['best_quality'])
+    acutal_config = ConfigBuilder().presets('best_quality').build()
+    assert acutal_config == expected_config
+
+    acutal_config = ConfigBuilder().presets(['best_quality']).build()
+    assert acutal_config == expected_config
+
+    expected_config = dict(presets=['best_quality', 'optimize_for_deployment'])
+    acutal_config = ConfigBuilder().presets(['best_quality', 'optimize_for_deployment']).build()
+    assert acutal_config == expected_config
+
+
+def test_presets_invalid_option():
+    with pytest.raises(AssertionError, match=r"The following preset are not recognized: .'unknown1'. - use one of the valid presets: .*"):
+        ConfigBuilder().presets('unknown1').build()
+
+    with pytest.raises(AssertionError, match=r"The following preset are not recognized: .'unknown2', 'unknown3'. - use one of the valid presets: .*"):
+        ConfigBuilder().presets(['best_quality', 'unknown2', 'unknown3']).build()
+
+
+def test_excluded_model_types():
+    expected_config = dict(excluded_model_types=['NN'])
+    acutal_config = ConfigBuilder().excluded_model_types('NN').build()
+    assert acutal_config == expected_config
+
+    acutal_config = ConfigBuilder().excluded_model_types(['NN']).build()
+    assert acutal_config == expected_config
+
+    expected_config = dict(excluded_model_types=['LR', 'NN'])
+    acutal_config = ConfigBuilder().excluded_model_types(['NN', 'LR']).build()
+    assert acutal_config == expected_config
+
+    expected_config = dict(excluded_model_types=['NN'])
+    acutal_config = ConfigBuilder().excluded_model_types(['NN', 'NN']).build()
+    assert acutal_config == expected_config
+
+
+def test_excluded_model_types_invalid_option():
+    with pytest.raises(AssertionError, match=r"unknown1 is not one of the valid models .*"):
+        ConfigBuilder().excluded_model_types('unknown1').build()
+
+    with pytest.raises(AssertionError, match=r"unknown2 is not one of the valid models .*"):
+        ConfigBuilder().excluded_model_types(['unknown2']).build()
+
+
+def test_included_model_types():
+    expected_config = dict(excluded_model_types=['RF', 'XT', 'KNN', 'GBM', 'CAT', 'XGB', 'QNN', 'LR', 'FASTAI', 'TRANSF', 'AG_TEXT_NN', 'AG_IMAGE_NN', 'FASTTEXT', 'VW'])
+    acutal_config = ConfigBuilder().included_model_types('NN').build()
+    assert acutal_config == expected_config
```

Comment:
nit: acutal -> actual, ditto for all other occurrences.

Reply:
fixed